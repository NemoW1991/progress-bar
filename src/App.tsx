import React, {useState} from 'react';
import { initialData } from './lib/constants';
import { getData } from './lib/api';
import { DataObj } from './lib/types';
import Main from './view';

export default function App() {

  const [data, setData] = useState<DataObj>(initialData);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  const fetchData = async() => {
    const dataSet = await getData();
    setIsLoading(true);
    setData(dataSet.data);
  }

  React.useEffect(() => {
    fetchData()
  },[]);

  return (
    <div className='app'>
      {isLoading && <Main data = {data} />}
    </div>
  )
}