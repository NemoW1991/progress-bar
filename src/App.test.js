import { shallow} from 'enzyme';
import App from './App';

describe('<App />', () => {
  let wrapper;
  beforeEach(() => 
    wrapper = shallow(<App />),
  );

  it('renders without crashing', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should render correctly', () => 
    expect(wrapper).toMatchSnapshot());  
});

