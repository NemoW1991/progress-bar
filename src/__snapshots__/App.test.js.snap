// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<App /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<View /> should render correctly 1`] = `ShallowWrapper {}`;
